app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/adapters/recycleradapters/FeedbackAdapter.kt

```diff
@@ -54,14 +54,18 @@ class FeedbackAdapter(val context: Context, val feedbackResponse: GetSkillFeedba
                         if (feedbackResponse.feedbackList[position].email != null &&
                                 !TextUtils.isEmpty(feedbackResponse.feedbackList[position].email)) {
                             Utils.setAvatar(context, feedbackResponse.feedbackList.get(position).avatar, holder.avatar)
-                            if (PrefManager.getToken() != null) {
-                                if (!feedbackResponse.feedbackList.get(position).email.equals(PrefManager.getStringSet(Constant.SAVED_EMAIL).iterator().next().toString(), true)) {
-                                    Utils.truncateEmailAtEnd(feedbackResponse.feedbackList.get(position).email)?.let { holder.feedbackEmail?.text = it }
+                            if (!feedbackResponse.feedbackList.get(position).userName.isNullOrEmpty()) {
+                                holder.feedbackEmail.text = feedbackResponse.feedbackList.get(position).userName
+                            } else {
+                                if (PrefManager.getToken() != null) {
+                                    if (!feedbackResponse.feedbackList.get(position).email.equals(PrefManager.getString(Constant.EMAIL, null), true)) {
+                                        Utils.truncateEmailAtEnd(feedbackResponse.feedbackList.get(position).email)?.let { holder.feedbackEmail?.text = it }
+                                    } else {
+                                        holder.feedbackEmail.text = feedbackResponse.feedbackList.get(position).email
+                                    }
                                 } else {
-                                    holder.feedbackEmail.text = feedbackResponse.feedbackList.get(position).email
+                                    Utils.truncateEmailAtEnd(feedbackResponse.feedbackList.get(position).email)?.let { holder.feedbackEmail?.text = it }
```

Comment:
This block is almost identical to one used in previous file. Please extract as a common function